packages/nexus/src/context.ts

```diff
@@ -0,0 +1,178 @@
+  taskQueue: string;
+}
+
+// Make it safe to use @temporalio/nexus with multiple versions installed.
+const asyncLocalStorageSymbol = Symbol.for('__temporal_nexus_context_storage__');
+if (!(globalThis as any)[asyncLocalStorageSymbol]) {
+  (globalThis as any)[asyncLocalStorageSymbol] = new AsyncLocalStorage<HandlerContext>();
+}
+
+export const asyncLocalStorage: AsyncLocalStorage<HandlerContext> = (globalThis as any)[asyncLocalStorageSymbol];
+
+function getHandlerContext(): HandlerContext {
+  const ctx = asyncLocalStorage.getStore();
+  if (ctx == null) {
+    throw new ReferenceError('Not in a Nexus handler context');
+  }
+  return ctx;
+}
+
+function getLogger() {
+  return getHandlerContext().log;
+}
+
+/**
+ * A logger for use in Nexus Handler scope.
+ */
+export const log: Logger = {
+  log(level: LogLevel, message: string, meta?: LogMetadata): any {
+    return getLogger().log(level, message, meta);
+  },
+  trace(message: string, meta?: LogMetadata): any {
+    return getLogger().trace(message, meta);
+  },
+  debug(message: string, meta?: LogMetadata): any {
+    return getLogger().debug(message, meta);
+  },
+  info(message: string, meta?: LogMetadata): any {
+    return getLogger().info(message, meta);
+  },
+  warn(message: string, meta?: LogMetadata): any {
+    return getLogger().warn(message, meta);
+  },
+  error(message: string, meta?: LogMetadata): any {
+    return getLogger().error(message, meta);
+  },
+};
+
+// TODO: also support getting a metrics handler.
+
+/**
+ * Returns a client to be used in a Nexus Operation's context, this Client is powered by the same NativeConnection that
+ * the worker was created with.
+ */
+export function getClient(): Client {
+  return getHandlerContext().client;
+}
+
+/**
+ * A handle to a running workflow that is returned by the {@link startWorkflow} helper.
+ * This handle should be returned by {@link WorkflowRunOperationHandler} implementations.
+ */
+export interface WorkflowHandle<_T> {
+  readonly workflowId: string;
+  readonly runId: string;
+}
+
+/**
+ * Options for starting a workflow using {@link startWorkflow}, this type is identical to the client's
+ * `WorkflowStartOptions` with the exception that `taskQueue` is optional and defaults to the current worker's task
+ * queue.
+ */
+export type WorkflowStartOptions<T extends Workflow> = Replace<ClientWorkflowStartOptions<T>, { taskQueue?: string }>;
+
+/**
+ * Starts a workflow run for a {@link WorkflowRunOperationHandler}, linking the execution chain to a Nexus Operation
+ * (subsequent runs started from continue-as-new and retries). Automatically propagates the callback, request ID, and
+ * back and forward links from the Nexus options to the Workflow.
+ */
+export async function startWorkflow<T extends Workflow>(
+  ctx: nexus.StartOperationContext,
+  workflowTypeOrFunc: string | T,
+  workflowOptions: WorkflowStartOptions<T>
+): Promise<WorkflowHandle<T>> {
+  const { client, taskQueue } = getHandlerContext();
```

Comment:
Users will need to be able to override with a client / task queue that is not the same as that inuse by the Nexus worker.